// js-logging/main1.js
'use strict';
//module.exports = require("./lib/index.js");

var Logging = require("./lib/index.js").Logging;
let asdf = new Logging();
asdf.warn('asdfasdf');
/*
function f1(str) {
	return str.toUpperCase();
}

let Logging = require('./lib/index.js');
let logger = new Logging({
	filters : [
	           f1, 'blue', //default filter
	           //the last item can be custom filter. here is "warn" and "error" filter
	           {
	        	   warning : 'yellow',
	        	   error : [f1, 'red' ]
	           }
	]
});

logger.debug('Hello World!');
logger.info('Hello World!');
logger.notice('Hello World!');
logger.warning('Hello World!');
logger.error('Hello World!');
logger.critical('Hello World!');
logger.alert('Hello World!');
logger.emergency('Hello World!');


let Logging1 = require('./lib/compatibility.js').console({
	filters : [
	           f1, 'green'
	]});

Logging1.debug('Hello World!');
Logging1.info('Hello World!');
Logging1.notice('Hello World!');
Logging1.warning('Hello World!');
Logging1.error('Hello World!');
Logging1.critical('Hello World!');
Logging1.alert('Hello World!');
Logging1.emergency('Hello World!');
*/
/*
"use strict";
var colors = require('colors/safe');
module.exports = function(conf) {
	return require('./console')({
		filters : {
			//log : do nothing
			trace : colors.magenta,
			debug : colors.cyan,
			info : colors.green,
			warn : colors.yellow,
			error : colors.red.bold
		}
	}, conf);
};
*/

/*
var logger2 = require('./lib/compatibility.js')
        .colorConsole(
                {
                    format : [
                              "${timestamp} <${title}> ${message} (in ${file}:${line})", //default format
                              {
                                  error : "${timestamp} <${title}> ${message} (in ${file}:${line})\nCall Stack:\n${stack}" // error format
                              }
                    ],
                    dateformat : "HH:MM:ss.L",
                    preprocess :  function(data){
                        data.title = data.title.toUpperCase();
                    }
                });
*/
/*
logger2.debug('Hello World!');
logger2.info('Hello World!');
logger2.notice('Hello World!');
logger2.warning('Hello World!');
*/
//logger2.error('Hello World!');
/*
logger2.critical('Hello World!');
logger2.alert('Hello World!');
logger2.emergency('Hello World!');
*/

/*
let Logging = require('./main.js');
  let logger = new Logging({
		transport : function(data) {
			console.log(data.output);
			return data;
		}
	});
	logger.close();
*/
let Logging1 = require('./js-logging.js').test;
Logging1();
let test = '';
